core/src/main/java/apoc/load/Xml.java

```diff
@@ -190,6 +194,7 @@ private boolean proceedReader(XMLStreamReader reader) throws XMLStreamException
     }
 
     private void handleNode(Deque<Map<String, Object>> stack, Node node, boolean simpleMode) {
+        terminationGuard.check();
```

Comment:
If this check fails does it throw an exception? do we need to do anything, or just let it happen :) perhaps a comment saying what it does could be useful

Reply:
Yes it throw an exception if the check fails. 
Do you mean add a comment for each piece of code with `terminationGuard.check()`?

However I personally don't think if it's really necessary add a comment, because it's already explained in the [org.neo4j.procedure.TerminationGuard interface](https://github.com/neo4j/neo4j/blob/5.1/community/procedure-api/src/main/java/org/neo4j/procedure/TerminationGuard.java#L29)

Reply:
Oh I didn't see the comment in the interface! This is fine then, thanks for explaining :)